### PR TITLE
Update Logic component in DevGuide + Update undo/redo implementation

### PR DESCRIPTION
**Pull request by Na-Nazhou**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -98,13 +98,13 @@ The `UI` component,
 image::LogicClassDiagram.png[]
```
Comment (kxyoke) [inline docs/DeveloperGuide.adoc]: Maybe before the picture can have a preface introducing the image

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -98,13 +98,13 @@ The `UI` component,
...
 *API* :
-link:{repoURL}/src/main/java/seedu/address/logic/Logic.java[`Logic.java`]
+link:{repoURL}/src/main/java/seedu/mark/logic/Logic.java[`Logic.java`]
 
-.  `Logic` uses the `AddressBookParser` class to parse the user command.
-.  This results in a `Command` object which is executed by the `LogicManager`.
-.  The command execution can affect the `Model` (e.g. adding a person).
-.  The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
-.  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
+*  `Logic` uses the `MarkParser` class to parse the user command.
```
Comment (kxyoke) [inline docs/DeveloperGuide.adoc]: Should have preface here also;
also need parallel structure

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -98,13 +98,13 @@ The `UI` component,
...
+*  This results in a `Command` object which is executed by the `LogicManager`.
+*  The command execution can affect the `Model` (e.g. adding a bookmark).
+*  The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
+*  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
 
 Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
```
Comment (kxyoke) [inline docs/DeveloperGuide.adoc]:
```suggestion
Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call as an example to show you how the `Logic` component works:
```

Comment (dorcastan) [reply docs/DeveloperGuide.adoc]: Building on @kxyoke's suggestion:

```suggestion
As an example of how the `Logic` component works, the following Sequence Diagram shows the interactions within the `Logic` component for the `execute("delete 1")` API call:
```

**Pull request merged**